dist/tools/pr_check/check.sh

```diff
@@ -40,21 +43,20 @@ SQUASH_COMMITS="$(git log $(git merge-base HEAD "${RIOT_MASTER}")...HEAD --prett
                   keyword_filter)"
 
 if [ -n "${SQUASH_COMMITS}" ]; then
-    echo -e "${CERROR}Pull request needs squashing:${CRESET}" 1>&2
-    echo -e "${SQUASH_COMMITS}"
+    if github_annotate_is_on; then
+        echo "${SQUASH_COMMITS}" | while read commit; do
+            github_annotate_warning_no_file \
```

Comment:
Why use a warning when this actually make the check to fail ?

Reply:
Because new contributors tend to immediately act on errors that they see in the CI, so I thought it would make more sense to keep it a warning. The check fail helps us to not merge accidentally, the message is a hint to the contributor.

Reply:
The default behavior, when run locally, is to fail. IMO, it makes more sense to keep the same.

Reply:
The check still fails. So it is kept the same.

Reply:
I meant that when run locally, the message is displayed using the "error" style:

https://github.com/RIOT-OS/RIOT/blob/acd846653a23f953a86635f886e302244cf78604/dist/tools/pr_check/check.sh#L50

That's why I think that, for consistency, the error style should also be used in the annotation.

Reply:
When a contributor runs the script locally, I assume they know what their doing. The reason I kept it a warning is, because I don't want a new contributor to squash commits because the Github Action tells them to.

Reply:
New contributors are supposed to have read the contributors guidelines where there's a note about "only squash when requested by reviewers". If they do that because of it, we just have to point them there and they won't do the "mistake" anymore.

Reply:
([for reference](https://github.com/RIOT-OS/RIOT/blob/master/CONTRIBUTING.md#squash-commits-after-review))

Reply:
Why add this extra step, if we can just do it by tooling?

Reply:
I guess the latest fixup is a good compromise.